spec/lib/active_versioning/model/versioned_spec.rb

```diff
@@ -0,0 +1,122 @@
+
+  subject { ActiveVersioning::Test::Post.create(title: 'So Post', body: 'Such interesting.  Very wow.', user: user) }
+
+  it { should have_many(:versions) }
+
+  describe "after create" do
+    it "creates a new version" do
+      expect {
+        ActiveVersioning::Test::Post.create(title: 'So Post', body: 'Such interesting.  Very wow.', user: user, version_author: user)
+      }.to change { ActiveVersioning::Test::Version.count }.by 1
+    end
+  end
+
+  describe "#version_author" do
+    it "should be able to set and view the author" do
+      subject.version_author = user
+
+      expect(subject.version_author).to eq user
+    end
+  end
+
+  describe "#live?" do
+    it { expect(subject.live?).to eq true }
+  end
+
+  describe "#version?" do
+    it { expect(subject.version?).to eq false }
+  end
+
+  describe "#current_draft" do
+    context "when the record is not persisted" do
+      subject { ActiveVersioning::Test::Post.new }
+
+      it "raises an error" do
+        expect { subject.current_draft }.to raise_error(ActiveVersioning::Model::Versioned::RecordNotPersisted)
+      end
+    end
+
+    context "when there are no draft versions" do
+      it "creates a new draft version wrapped in a VersionProxy" do
+        expect {
+          subject.current_draft
+
+          # Verify we have a VersionProxy
+          expect(subject.current_draft.__getobj__).to eq subject
+        }.to change { subject.versions.count }.by 1
+      end
+    end
+
+    context "when there are existing draft versions" do
+      let!(:draft) { subject.current_draft }
+
+      it "grabs the first draft version and wraps it in a VersionProxy" do
+        expect {
+          expect(subject.current_draft).to eq draft
+
+          # Verify we have a VersionProxy
+          expect(draft.__getobj__).to eq subject
+        }.to_not change { subject.versions.count }
+      end
+    end
+
+    it "is not live" do
+      expect(subject.current_draft.live?).to eq false
+    end
+
+    it "is a version" do
+      expect(subject.current_draft.version?).to eq true
+    end
+  end
+
+  describe "#current_draft?" do
+    context "when the record has a draft version" do
+      before { subject.current_draft(true) }
+
+      it { expect(subject.current_draft?).to eq true }
+    end
+
+    context "when the record does not have a draft version" do
+      it { expect(subject.current_draft?).to eq false }
+    end
+  end
+
+  describe "#destroy_draft" do
+    before { subject.current_draft(true) }
+
+    it "destroys all draft versions" do
+      expect { subject.destroy_draft }.to change { subject.current_draft? }.from(true).to(false)
+    end
+  end
+
+  describe "#create_draft_from_version" do
+    let!(:version) { subject.current_draft }
+
+    before { version.commit }
+
+    it "creates a new draft from the given version" do
+      expect {
+        subject.create_draft_from_version(version.id)
+      }.to change { subject.versions.count }.by 1
```

Comment:
this test is really just checking that a new version is created, is there anything else worth checking to ensure that the draft was created from the given version?